renovate/create-wrapper.sh

```diff
@@ -0,0 +1,75 @@
+#!/bin/bash
+set -euo pipefail
+
+# Script to create a Renovate wrapper that auto-exports RENOVATE_* secrets
+# Usage: create-wrapper.sh <output-path>
+# Note: Expects SECRETS_JSON environment variable to be set
+
+OUTPUT_PATH="$1"
+
+echo "🔧 Creating Renovate wrapper script..."
+
+# Validate that SECRETS_JSON is set
+if [ -z "${SECRETS_JSON:-}" ]; then
+  echo "❌ Error: SECRETS_JSON environment variable is not set"
+  exit 1
+fi
+
+# Clean and validate the JSON using jq (this removes control characters)
+echo "🔍 Cleaning and validating JSON..."
+CLEANED_JSON=$(echo "$SECRETS_JSON" | jq -c . 2>&1) || {
+  echo "❌ Error: Invalid JSON in SECRETS_JSON"
+  exit 1
+}
+
+echo "✅ JSON cleaned and validated"
+
+# Encode to base64 (use -w 0 to disable line wrapping)
+SECRETS_BASE64=$(echo "$CLEANED_JSON" | base64 -w 0)
+
+echo "✅ Secrets validated and encoded"
+
+# Create the wrapper script with base64-encoded secrets
+cat > "$OUTPUT_PATH" <<WRAPPER_EOF
+#!/bin/bash
+set -euo pipefail
+
+# Decode secrets from base64
+ALL_SECRETS=\$(echo "${SECRETS_BASE64}" | base64 -d)
+
+echo "🔧 Auto-detecting and exporting RENOVATE_* secrets..."
+echo ""
+
+# Track count of exported secrets
+EXPORTED_COUNT=0
+
+# Export each RENOVATE_* secret
+while IFS='=' read -r key value; do
+  if [ -n "\$key" ] && [ -n "\$value" ]; then
+    export "\$key=\$value"
+    EXPORTED_COUNT=\$((EXPORTED_COUNT + 1))
+    
+    # Mask sensitive values in output
+    if [[ \$key == *"PASSWORD"* ]] || [[ \$key == *"TOKEN"* ]] || [[ \$key == *"KEY"* ]]; then
+      echo "  ✓ \$key=***"
+    else
+      echo "  ✓ \$key=\$value"
+    fi
+  fi
+done < <(echo "\$ALL_SECRETS" | jq -r 'to_entries[] | select(.key | startswith("RENOVATE_")) | "\(.key)=\(.value)"')
+
+echo ""
+echo "✅ Exported \$EXPORTED_COUNT RENOVATE_* secrets"
+echo "✅ Configuration complete"
+echo ""
+echo "🚀 Starting Renovate..."
+echo ""
+
+exec renovate "\$@"
+WRAPPER_EOF
+
+# Make executable
+chmod +x "$OUTPUT_PATH"
+
+echo "✅ Wrapper script created at: $OUTPUT_PATH"
+echo ""
```

Comment:
Instead of a Bash script that creates a wrapper, which is tricky to read and maintain, should we just parse the JSON with Python and use Python to call the Renovate process with the appropriate variables set? That sounds like it'd be much easier to understand and maintain.